File: backend/app/schemas/task.py
"""
任务执行Pydantic模式
"""
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime
from app.models.task_execution import TaskStatus


class TaskExecutionBase(BaseModel):
    """任务执行基础模式"""
    task_name: str = Field(..., description="任务名称")
    task_type: str = Field(..., description="任务类型")
    status: TaskStatus = Field(..., description="执行状态")
    start_time: datetime = Field(..., description="开始时间")
    end_time: Optional[datetime] = Field(None, description="结束时间")
    duration: Optional[str] = Field(None, description="执行时长")
    result: Optional[Dict[str, Any]] = Field(None, description="执行结果详情")
    error_message: Optional[str] = Field(None, description="错误信息")
    triggered_by: str = Field(..., description="触发方式")
    target_date: Optional[str] = Field(None, description="目标日期")


class TaskExecutionResponse(TaskExecutionBase):
    """任务执行响应"""
    id: int
    created_at: datetime
    
    class Config:
        from_attributes = True


class TaskExecutionListResponse(BaseModel):
    """任务执行列表响应"""
    items: List[TaskExecutionResponse]
    total: int
    page: int
    page_size: int
    total_pages: int


class TaskExecutionCreate(BaseModel):
    """创建任务执行记录"""
    task_name: str
    task_type: str
    triggered_by: str = "manual"
    target_date: Optional[str] = None


class TaskRunRequest(BaseModel):
    """手动执行任务请求"""
    task_types: Optional[List[str]] = Field(None, description="任务类型列表，不指定则执行所有任务")
    target_date: Optional[str] = Field(None, description="目标日期，格式：YYYY-MM-DD，不指定则使用交易日")


class TaskRunResponse(BaseModel):
    """任务执行响应"""
    execution_id: int
    message: str
    task_types: List[str]

File: frontend/src/api/task.ts
import apiClient from './client'

export type TaskStatus = 'pending' | 'running' | 'success' | 'failed'

export interface TaskExecution {
  id: number
  task_name: string
  task_type: string
  status: TaskStatus
  start_time: string
  end_time?: string
  duration?: string
  result?: {
    success_count?: number
    total_count?: number
    task_results?: Record<string, any>
  }
  error_message?: string
  triggered_by: string
  target_date?: string
  created_at: string
}

export interface TaskExecutionListParams {
  page?: number
  page_size?: number
  task_type?: string
  status?: TaskStatus
  task_name?: string
}

export interface TaskExecutionListResponse {
  items: TaskExecution[]
  total: number
  page: number
  page_size: number
  total_pages: number
}

export interface TaskStatusSummary {
  [taskType: string]: {
    task_name: string
    status: TaskStatus
    last_run_time?: string
    last_success_time?: string
    duration?: string
    error_message?: string
  }
}

export interface TaskRunRequest {
  task_types?: string[]
  target_date?: string
}

export interface TaskRunResponse {
  execution_id: number
  message: string
  task_types: string[]
}

export interface TaskTypesResponse {
  task_types: string[]
  task_names: Record<string, string>
}

export const taskApi = {
  getExecutions: async (params: TaskExecutionListParams): Promise<TaskExecutionListResponse> => {
    const cleanParams: Record<string, any> = {}
    if (params.page) cleanParams.page = params.page
    if (params.page_size) cleanParams.page_size = params.page_size
    if (params.task_type) cleanParams.task_type = params.task_type
    if (params.status) cleanParams.status = params.status
    if (params.task_name) cleanParams.task_name = params.task_name
    
    return await apiClient.get('/tasks/executions', { params: cleanParams })
  },

  getExecutionById: async (id: number): Promise<TaskExecution> => {
    return await apiClient.get(`/tasks/executions/${id}`)
  },

  getStatusSummary: async (): Promise<TaskStatusSummary> => {
    return await apiClient.get('/tasks/status')
  },

  runTasks: async (data: TaskRunRequest): Promise<TaskRunResponse> => {
    return await apiClient.post('/tasks/run', data)
  },

  getTaskTypes: async (): Promise<TaskTypesResponse> => {
    return await apiClient.get('/tasks/task-types')
  },
}

